pkg/scheduler/framework/interface.go

```diff
@@ -199,6 +202,9 @@ func (s *Status) FailedPlugin() string {
 
 // Reasons returns reasons of the Status.
 func (s *Status) Reasons() []string {
+	if s.err != nil {
+		return append(s.reasons, s.err.Error())
```

Comment:
Should we add the error message at head.

Reply:
Currently, in all cases,when `s.err!=nil`, s.reasons will be nil.
So head or tail is basiclly same. We can leave a discussion here.

Reply:
In theory, it can happen. But I'm ok with this now until something really happens.